azure-pipelines.yml

```diff
@@ -105,7 +105,7 @@ jobs:
 
   - bash: |
       fork_origin=$(git merge-base --fork-point origin/master)
-      changed_files=$(git diff --name-only $fork_origin | grep -v /vendor/ | grep -E '\.(cpp|cc|c|h)$')
+      changed_files=$(git diff --name-only $fork_origin | grep -v '^vendor' | grep -E '\.(cpp|cc|c|h)$')
```

Comment:
Decided to alter this so it'll only remove things where vendor is at the beginning of the path.

With the plain 'vendor' any file that has 'vendor' in the path or filename will get left out of lint. This could cause a file that isn't vendored into the repository to be left out of lint CI.

A rule that we don't put 'vendor' on any path or filename that isn't vendored is reasonable but the issue was if we fail to apply that rule then we'll end up with a file in the repository not getting the lint check and we won't get anything that flags that up to us.

This way round if we want something that's vendored that isn't under the top-level vendor/ we'll see a CI failure and can consider the best way forward.

May be worth altering this to '^vendor/' to make it strictly only filter stuff in the vendor/ directory.

Reply:
This rule was copied from OpenTitan, where vendor directories are all over the place. But for Ibex LGTM.

Reply:
Interesting, surprised the /vendor/ style works in OT but not in Ibex which is why I switched to 'vendor' (this PR was failing CI because it was linting the vendored coremark source as the grep had failed to filter things with vendor in).